### ISSUE TITLE
AS cancelling wrapped jobs before mpirun ?

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#663]
In GitLab by @etiennesky on Feb 19, 2021, 14:02

Hi @mcastril @dbeltrankyl 

using AS 0.12 fixed.

I have been having this recurring issue since yesterday, some auto-ecearth3 wrapped SIM jobs are failing in mpirun.

Any help is appreciated.

e.g. /esarchive/autosubmit/a3h6/tmp/LOG_a3h6/a3h6_18500101_fc0_57_SIM.cmd.0.err (which mysteriously has not been transferred back to esarchive, I copied it myself). 

```
+ mpirun -machinefile machinefile -np 2 ./xios_server.exe : -np 229 ./nemo.exe : -np 256 ./ifsmaster-ecconf -v ecmwf -e a3h6 : -np 8 ./guess_T255 guess.ins -parallel : -np 4 ./appl-tm5-co2.x tm5-run.rc : -np 1 ./runoff-mapper.exe
srun: error: SLURM job 14309532 has expired
srun: Check SLURM_JOB_ID environment variable. Expired or invalid job 14309532
```

and in /esarchive/autosubmit/a3h6/tmp/20210218_130050_run.log around the same time (Feb 19 00:52)

```

2021-02-19 00:52:17,796 Job a3h6_18500101_fc0_57_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_58_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_59_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_60_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_61_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_62_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_63_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_64_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_65_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_66_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_67_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_68_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_69_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,797 Job a3h6_18500101_fc0_70_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,798 Job a3h6_18500101_fc0_71_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,798 Job a3h6_18500101_fc0_72_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,798 Job a3h6_18500101_fc0_73_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,798 Job a3h6_18500101_fc0_74_SIM is SUBMITTED and waiting for dependencies
2021-02-19 00:52:17,798 Job a3h6_18500101_fc0_75_SIM is SUBMITTED and waiting for dependencies
[ERROR] 2021-02-19 00:52:17,798 It seems there are no inner jobs running in the wrapper. Cancelling...
2021-02-19 00:52:17,798 Cancelling job with id 14309532
[WARNING] 2021-02-19 00:52:21,222 Command scancel 14309532 in mn2.bsc.es warning: /home/cns04/cns04886/.bashrc: line 54: ppwd: command not found

2021-02-19 00:52:21,222 Command scancel 14309532 in mn2.bsc.es successful with out message: 

```

according to the user portal, the job was cancelled (by AS supposedly) after running 17 seconds

and I got a similar problem with expid a3hg but there are not logs, so it was cancelled before it was even ran.

e.g. in /esarchive/autosubmit/a3hg/tmp/20210217_210919_run.log around (2021-02-19 01:04:50)

```
2021-02-19 01:04:50,431 Job a3hg_18500101_fc0_53_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,431 Job a3hg_18500101_fc0_54_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,431 Job a3hg_18500101_fc0_55_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,431 Job a3hg_18500101_fc0_56_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_57_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_58_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_59_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_60_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_61_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_62_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_63_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_64_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_65_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_66_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_67_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_68_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,432 Job a3hg_18500101_fc0_69_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_70_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_71_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_72_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_73_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_74_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_75_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_76_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_77_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_78_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_79_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_80_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_81_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,433 Job a3hg_18500101_fc0_82_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,434 Job a3hg_18500101_fc0_83_SIM is SUBMITTED and waiting for dependencies
2021-02-19 01:04:50,434 Job a3hg_18500101_fc0_84_SIM is SUBMITTED and waiting for dependencies
[ERROR] 2021-02-19 01:04:50,434 It seems there are no inner jobs running in the wrapper. Cancelling...
2021-02-19 01:04:50,434 Cancelling job with id 14310786
[WARNING] 2021-02-19 01:04:53,419 Command scancel 14310786 in mn1.bsc.es warning: /home/cns04/cns04886/.bashrc: line 54: ppwd: command not found

2021-02-19 01:04:53,420 Command scancel 14310786 in mn1.bsc.es successful with out message: 
2021-02-19 01:04:53,435 Updating FAILED jobs

```

according to the user portal, the job was cancelled (by AS supposedly) after running 18 seconds

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 14:07

Hi @etiennesky ,

I didn't touch 3.12.0 version so, it should be some related to your environment or the platform itself,

I notice that there is shown an issue with a command that you have inside the .bashrc, perhaps that is messing up something and the slurm variables or some doesn't load up properly? 

I say this for:

 `[WARNING] 2021-02-19 01:04:53,419 Command scancel 14310786 in mn1.bsc.es warning: /home/cns04/cns04886/.bashrc: line 54: ppwd: command not found`

seems that it is unable to recognize bash commands, 

Did you tried use 3.13.0 to see if it also happen with that version?

**Comment by kinow** [comment]
In GitLab by @etiennesky on Feb 19, 2021, 15:20

Hi @dbeltrankyl don't worry about the ppwd warning, it is harmless and is always in my logs.

I have been running something like 20 very long experiments in the last months with the same setup and AS version, and never had this problem.

I am wondering if there is a connectivity issue ?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 15:43

> I am wondering if there is a connectivity issue ?

That doesn't seem the case because autosubmit is able to retrieve the ID of the job however I'm not sure and I didn't receive any issue like this from other AS users

Do you have any idea @mcastril , @wuruchi ?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 15:44

@etiennesky  Perhaps you can change the machine from where you are launching the jobs, because yesterday an outage affected autosubmit machines for while ( although I think it was only the 02)

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:11

> I have been running something like 20 very long experiments in the last months with the same setup and AS version, and never had this problem.
> 
> I am wondering if there is a connectivity issue ?

If it was a connectivity issue then there's not much we can do on the Autosubmit side. 3.12.0 was patched to 3.12.0-fixed but nothing more intrusive can be done in this version. 3.13.0b will be out very soon.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:13

@etiennesky, given the machinefile warnings mentioned by Operations and the sudden model crash, could it be some problem of resources? Have you been running using this specific distribution?

**Comment by kinow** [comment]
In GitLab by @etiennesky on Feb 19, 2021, 16:27

Hi I understand the concern about the machinefiles... somehow I made a mistake in the configuration of my C4MIP experiments which all have run with this setup, so it's not the issue.

Besides, the error happens even before the model starts running, and it just happened since yesterday...

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:36

That was my impression, that `mpirun` had been invoked but the model couldn't even run. 

The wrapper log shows:

```
slurmstepd: error: *** JOB 14309532 ON s01r1b47 CANCELLED AT 2021-02-19T00:52:37 ***
```                                                                                        

So it's clear that it was a cancellation, as Operations confirmed. Then it had to come from either Autosubmit or the scheduler (sometimes errors produce cancellations).

If it was Autosubmit, it was very timely, to do it right after the `mpirun` and before the model starts to run. That was why I was suspecting on the machine side (not if it had been cancelled before the `mpirun` or during the model run). But as you know, the model start is a critic moment, where most of the errors take place. Could you look at the portal in case that any of the nodes had some memory already occupied?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:38

Ok I saw the previous one was cancelled too, but at the time of the modules loading. 

@dbeltrankyl , is there any circumstance in which Autosubmit may cancel a job in 3.12.0, besides exceeding inner job wallclock?

**Comment by kinow** [comment]
In GitLab by @etiennesky on Feb 19, 2021, 16:39

either no graphs or "There is no data available for this metric."

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 16:51

If inner_job is failed, unknown,wallclock:

`cancel_failed_wrapper_job `

If there are no inner_jobs running but the wrapper job running (this issue?)

this dict is filled with
```
                                Log.info("Job {0} started at {1}".format(jobname, str(parse_date(start_time))))
                                self.running_jobs_start[job] = start_time
```

If a job has a failing queuing status :
```
                if self._queuing_reason_cancel(reason):
                    Log.error("Job {0} will be cancelled and set to FAILED as it was queuing due to {1}", self.name, reason)
```

@etiennesky , With this in mind, could you try to increase the safety time or the CHECK_TIME_WRAPPER variable?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:56

> unknown

So does it mean that if `sacct` or `ssh` connections are not returning an output then the wrapper could be cancelled?

Of course, this is not a usual situation because when `sacct` is not available then it's likely that `scancel` is not available too. But a problem with just the sacct or the internal log ids could provoke this situation right?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 17:01

> failed

And this recalled me a recurring idea / question that I had in mind.

https://earth.bsc.es/gitlab/es/autosubmit/-/issues/664

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 17:01

Could happen but only in the case that in one iteration, the wrapper passed from QUEUING into RUNNING and autosubmit failed to check the inner jobs for the first time, if it fails when it already recognized some inner job it won't cancel the job.

Although this is a quick view and maybe I missed something from a old code. 3.13.0b doesn't work like this anymore

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 17:18

I am doing a search to know if someone else had the same problem these days.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 18:43

In the AS local run logs, I see some signs that there might be a recent issue in MN4. From 95 errors in 3 years time, these are from these month:

```
a1ua/tmp/20210217_140205_run.log:[ERROR] 2021-02-18 11:13:37,621 It seems there are no inner jobs running in the wrapper. Cancelling...
a1ua/tmp/20210217_140205_run.log:[ERROR] 2021-02-18 13:44:35,386 It seems there are no inner jobs running in the wrapper. Cancelling...
a1ua/tmp/20210217_140205_run.log:[ERROR] 2021-02-18 13:49:01,285 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_111923_run.log:[ERROR] 2021-02-18 11:38:07,008 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_111923_run.log:[ERROR] 2021-02-18 12:00:32,850 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_111923_run.log:[ERROR] 2021-02-18 13:48:54,392 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_192652_run.log:[ERROR] 2021-02-18 23:38:29,863 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_192652_run.log:[ERROR] 2021-02-19 05:42:30,716 It seems there are no inner jobs running in the wrapper. Cancelling...
a27e/tmp/20210218_192652_run.log:[ERROR] 2021-02-19 05:50:45,108 It seems there are no inner jobs running in the wrapper. Cancelling...
a369/tmp/20210216_104224_run.log:[ERROR] 2021-02-18 14:06:54,053 It seems there are no inner jobs running in the wrapper. Cancelling...
a3dv/tmp/20210202_193140_run.log:[ERROR] 2021-02-19 05:26:02,780 It seems there are no inner jobs running in the wrapper. Cancelling...
a3dv/tmp/20210202_193140_run.log:[ERROR] 2021-02-19 05:46:38,587 It seems there are no inner jobs running in the wrapper. Cancelling...
a3e8/tmp/20210201_084136_run.log:[ERROR] 2021-02-03 00:09:11,631 It seems there are no inner jobs running in the wrapper. Cancelling...
a3fr/tmp/20210213_084631_run.log:[ERROR] 2021-02-16 21:57:17,795 It seems there are no inner jobs running in the wrapper. Cancelling...
a3fy/tmp/20210202_181818_run.log:[ERROR] 2021-02-02 19:32:53,603 It seems there are no inner jobs running in the wrapper. Cancelling...
a3h6/tmp/20210210_100838_run.log:[ERROR] 2021-02-17 23:23:11,609 It seems there are no inner jobs running in the wrapper. Cancelling...
a3h6/tmp/20210218_130050_run.log:[ERROR] 2021-02-19 00:52:17,798 It seems there are no inner jobs running in the wrapper. Cancelling...
a3h9/tmp/20210209_114850_run.log:[ERROR] 2021-02-10 13:22:00,077 It seems there are no inner jobs running in the wrapper. Cancelling...
a3h9/tmp/20210209_114850_run.log:[ERROR] 2021-02-10 15:39:18,584 It seems there are no inner jobs running in the wrapper. Cancelling...
a3he/tmp/20210217_205037_run.log:[ERROR] 2021-02-18 18:42:16,203 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hf/tmp/20210217_210120_run.log:[ERROR] 2021-02-18 18:44:31,950 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hg/tmp/20210217_210919_run.log:[ERROR] 2021-02-19 01:04:50,434 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hh/tmp/20210210_122850_run.log:[ERROR] 2021-02-10 16:09:59,466 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hh/tmp/20210210_122850_run.log:[ERROR] 2021-02-10 16:24:46,796 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hh/tmp/20210210_122850_run.log:[ERROR] 2021-02-10 18:00:07,631 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hh/tmp/20210210_122850_run.log:[ERROR] 2021-02-10 18:49:12,073 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hi/tmp/20210210_122859_run.log:[ERROR] 2021-02-10 15:12:50,458 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hi/tmp/20210210_122859_run.log:[ERROR] 2021-02-10 16:09:59,138 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hj/tmp/20210210_122922_run.log:[ERROR] 2021-02-10 15:12:48,530 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hj/tmp/20210210_122922_run.log:[ERROR] 2021-02-10 16:09:59,249 It seems there are no inner jobs running in the wrapper. Cancelling...
a3hj/tmp/20210210_122922_run.log:[ERROR] 2021-02-10 16:24:55,374 It seems there are no inner jobs running in the wrapper. Cancelling...
```
31 cases.

These may be real job failures, detected by AS, but 31/95 for 19 days it's a lot. I cannot verify this information with the remote inner job logs because some of them are in RES/PRACE accounts, but I could confirm at least two cases from Maria and Joan.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 18:56

@dbeltrankyl , could it be that the remote STAT cannot be retrieved because of any I/O issue?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 22, 2021, 09:16

I believe that it happens because the file isn't generated yet, sometimes it also happens to 3.13.0b for that reason there is a retrial mechanism to ensure that it wasn't a false negative.

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 3, 2021, 15:22

We consider the issue solved in 3.13.0:

* There is a retrial for the process considering a wrapper failed to avoid false negatives.
* Failed jobs should be retried within the same wrapper (without canceling it).
